packages/twenty-front/src/modules/apollo/hooks/useCachedRootQuery.ts

```diff
@@ -0,0 +1,49 @@
+import { useApolloClient } from '@apollo/client/react/hooks/useApolloClient';
+import gql from 'graphql-tag';
+
+import { useMapFieldMetadataToGraphQLQuery } from '@/object-metadata/hooks/useMapFieldMetadataToGraphQLQuery';
+import { ObjectMetadataItem } from '@/object-metadata/types/ObjectMetadataItem';
+
+export const useCachedRootQuery = ({
+  objectMetadataItem,
+  isArrayOfRecords,
```

Comment:
not a big fan if isArrayOfRecords, we should stick to the query api: { query: findMany | findOne }. We would likely also pass query variables at some point